frontend/src/components/filter/FilterMenu.js

```diff
@@ -54,27 +65,43 @@ export default function FilterMenu({ filters, onApplyFilters }) {
   };
 
   // reset state if we hit cancel
-  const onCancel = () => setItems([...filters]);
+  const onCancel = () => {
+    const copyOfFilters = filters.map((filter) => ({ ...filter }));
+    setItems(copyOfFilters);
+  };
 
-  const onUpdateFilter = (id, name, value, toggleAllChecked) => {
-    const newItems = [...items];
+  const onUpdateFilter = (id, name, value) => {
+    //
+    // this is bonkers... we need to do more than map the array
+    // what escaped me originally was that just an array spread creates a new
+    // array of references... to the same objects as before
+    // therefore, this function was mutating state in unexpected ways
+    //
+    // hence this real humdinger of a line of javascript
```

Comment:
:upside_down_face: Another option is https://lodash.com/docs/4.17.15#cloneDeep (I don't really have a preference, I think this is fine as is)